plugins/inputs/docker/docker.go

```diff
@@ -434,8 +434,7 @@ func (d *Docker) gatherContainer(
 	var cname string
 	for _, name := range container.Names {
 		trimmedName := strings.TrimPrefix(name, "/")
-		match := d.containerFilter.Match(trimmedName)
-		if match {
+		if len(strings.Split(trimmedName, "/")) == 1 {
```

Comment:
This isn't quite the same, since it will also match strings with multiple `/` characters. The behavior here is intentional to filter out container names from links.

Using the example from https://docs.docker.com/network/links/#communication-across-links, the containers would have the following names:

| Container | Names |
|-|-|
| `web` | `/web` |
| `db` | `/db`, `/web/db` |

Reply:
If it would be helpful, I can add a comment similar to the one in the Docker CLI:
https://github.com/docker/cli/blob/cde469bf1aad14146d57710205399899e465aadc/cli/command/formatter/container.go#L124-L126

Reply:
Ah sorry, I misread this. We've already trimmed the name here. So my logic is actually wrong.

I think what we really want is
```suggestion
		if !strings.Contains(trimmedName, "/") {
```